source/dofs/dof_handler_policy.cc

```diff
@@ -280,6 +280,80 @@ namespace internal
               return numbers::invalid_unsigned_int;
             }
         }
+
+
+
+        /**
+         * Same as above, but for vertices.
+         */
+        template <int dim, int spacedim>
+        unsigned int
+        get_most_dominating_fe_index_on_vertex(
+          const hp::DoFHandler<dim, spacedim> &dof_handler,
+          const unsigned int                   vertex_index)
+        {
+          const unsigned int n_active_fe_indices =
+            dealii::internal::DoFAccessorImplementation::Implementation::
+              n_active_vertex_fe_indices(dof_handler, vertex_index);
+
+          unsigned int dominating_fe_index = 0;
+          for (; dominating_fe_index < n_active_fe_indices;
+               ++dominating_fe_index)
```

Comment:
The following many lines are basically just a copy of the function above. Can you refactor this in the following way:
* Create a function `hp::FECollection::find_most_face_dominating_fe()` that takes a `std::set` and that basically has the logic of the function here and the function above
* Call this function from here and above; all you will then have to do is collect the set of active fe indices

Reply:
I agree, but I won't make this a member of `FECollection`, since I don"t want to find the dominating FE on the whole FECollection, but just a subset. Working on it!

Reply:
I know, but the `find_least_face_dominating_fe()` function `hp::FECollection` also already takes a `std::set` of fe indices. I think the function I propose above would fit this pattern just fine!